solana/programs/token-bridge-relayer/src/processor/inbound.rs

```diff
@@ -0,0 +1,328 @@
+    wormhole::{self, program::Wormhole},
+};
+
+type PostedRelayerMessage = token_bridge::PostedTransferWith<RelayerMessage>;
+
+#[derive(Accounts)]
+#[instruction(vaa_hash: [u8; 32])]
+pub struct CompleteTransfer<'info> {
+    /// Payer will pay for completing the Wormhole transfer tokens and create temporary
+    /// token account.
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// This program's config.
+    #[account(
+        seeds = [TbrConfigState::SEED_PREFIX],
+        bump
+    )]
+    pub tbr_config: Box<Account<'info, TbrConfigState>>,
+
+    /// Mint info. This is the SPL token that will be bridged over to the
+    /// foreign contract. Mutable.
+    ///
+    /// In the case of a native transfer, it's the mint for the token wrapped by Wormhole;
+    /// in the case of a wrapped transfer, it's the native SPL token mint.
+    #[account(mut)]
+    pub mint: Account<'info, Mint>,
+
+    /// Recipient associated token account. The recipient authority check
+    /// is necessary to ensure that the recipient is the intended recipient
+    /// of the bridged tokens. Mutable.
+    #[account(
+        mut,
+        associated_token::mint = mint,
+        associated_token::authority = recipient
+    )]
+    pub recipient_token_account: Box<Account<'info, TokenAccount>>,
+
+    /// CHECK: recipient may differ from payer if a relayer paid for this
+    /// transaction. This instruction verifies that the recipient key
+    /// passed in this context matches the intended recipient in the vaa.
+    #[account(mut)]
+    pub recipient: AccountInfo<'info>,
+
+    /// Verified Wormhole message account. The Wormhole program verified
+    /// signatures and posted the account data here. Read-only.
+    #[account(
+        seeds = [
+            wormhole::SEED_PREFIX_POSTED_VAA,
+            &vaa_hash
+        ],
+        seeds::program = wormhole_program.key(),
+        bump,
+        constraint = vaa.data().to() == crate::ID @ TokenBridgeRelayerError::InvalidTransferToAddress,
+        constraint = vaa.data().to_chain() == wormhole::CHAIN_ID_SOLANA @ TokenBridgeRelayerError::InvalidTransferToChain,
+        constraint = vaa.data().token_chain() == wormhole::CHAIN_ID_SOLANA @ TokenBridgeRelayerError::InvalidTransferTokenChain
+    )]
+    pub vaa: Account<'info, PostedRelayerMessage>,
+
+    /// Program's temporary token account. This account is created before the
+    /// instruction is invoked to temporarily take custody of the payer's
+    /// tokens. When the tokens are finally bridged in, the tokens will be
+    /// transferred to the destination token accounts. This account will have
+    /// zero balance and can be closed.
+    #[account(
+        init,
+        payer = payer,
+        seeds = [
+            SEED_PREFIX_TEMPORARY,
+            mint.key().as_ref(),
+        ],
+        bump,
+        token::mint = mint,
+        token::authority = tbr_config
+    )]
+    pub temporary_account: Account<'info, TokenAccount>,
```

Comment:
The "temporary" token account is never closed.

Reply:
Yes it is, see the end of the function.

Reply:
It's there now because [Solano and I added it yesterday](https://github.com/XLabs/arbitrary-token-transfers/commit/61c8e644228e9bb5fffe9bb279aa0d6bb58e1393).

Reply:
This was [the end of the function at the time](https://github.com/real-felix/arbitrary-token-transfers/blob/4c15cfaa2146790f7cab311ec1fa296676d155ac/solana/programs/token-bridge-relayer/src/processor/inbound.rs#L212).